### PR TITLE
Release/5.0.0

### PR DESCRIPTION
**Pull request by tonidero**
#### Apple AdServices support
New method for automatic collecting of attribution tokens on iOS and macOS using AdServices - `Purchases.enableAdServicesAttributionTokenCollection()`

#### StoreKit 2 support
This version of the SDK automatically uses StoreKit 2 APIs under the hood only for APIs that the RevenueCat team has determined work better than StoreKit 1.

#### New types and cleaned up naming
New types that wrap native types from Apple, Google and Amazon, and we cleaned up the naming of other types and methods for a more consistent experience.

### Removed APIs
- `setUp` has been removed in favor of `configure`
- `configure` now expects an Object for the different parameters. Example:
```javascript
Purchases.configure({
    apiKey: "key",
    appUserID: "user ID",
    observerMode: false,
    userDefaultsSuiteName: "suite name",
    usesStoreKit2IfAvailable: true,
    useAmazon: true
});
```
- `identify` and `createAlias` have been removed in favor of `logIn`.
- `reset` has been removed in favor of `logOut`.
- `addAttributionData` has been removed in favor of `set<NetworkID> methods`.
- `PurchasesStoreProduct`: removed `intro_price_string`, `intro_price_period`, `intro_price_cycles`, `intro_price_period_unit`, `intro_price_period_number_of_units` in favor of new `introPrice: PurchasesIntroPrice`.
- `PurchasesStoreTransaction`: removed `revenueCatId` and `productId` in favor of `transactionIdentifier` and `productIdentifier` respectively.

### Renamed APIs

| 4.x | 5.0.0 |
| :-: | :-: |
| `PurchaserInfo` | `CustomerInfo` |
| `PurchasesProduct` | `PurchasesStoreProduct` |
| `PurchasesStoreProductProduct.price_string` | `PurchasesStoreProductProduct.priceString` |
| `PurchasesStoreProductProduct.currency_code` | `PurchasesStoreProductProduct.currencyCode` |
| `PurchasesTransaction` | `PurchasesStoreTransaction` |
| `PurchasesTransaction.revenueCatId` | `PurchasesStoreTransaction.transactionIdentifier` |
| `PurchasesTransaction.productId` | `PurchasesStoreTransaction.productIdentifier` |
| `PurchasesDiscount` | `PurchasesStoreProductDiscount` |
| `PurchasesPaymentDiscount` | `PurchasesPromotionalOffer` |
| `Purchases.restoreTransactions` | `Purchases.restorePurchases` |
| `Purchases.getPaymentDiscount` | `Purchases.getPromotionalOffer` |
| `Purchases.invalidatePurchaserInfoCache` | `Purchases.invalidateCustomerInfoCache` |
| `Purchases.addPurchaserInfoUpdateListener` | `Purchases.addCustomerInfoUpdateListener` |
| `Purchases.removePurchaserInfoUpdateListener` | `Purchases.removeCustomerInfoUpdateListener` |

### CODE REVIEW

File: .version [hunk 1]
```diff
@@ -1 +1 @@
-5.0.0-beta.6
```
Comment (NachoSoto) [inline .version:1]: 😍

File: CHANGELOG.latest.md [hunk 2]
```diff
@@ -1,3 +1,47 @@
-⚠️⚠️ This is a pre-release version. ⚠️⚠️
+#### Apple AdServices support
```
Comment (NachoSoto) [inline CHANGELOG.latest.md:1]: Having this first makes it seem like it's the main change in the release. Maybe we should move this down?

**Review by NachoSoto — APPROVED** [review]
🥳

File: CHANGELOG.latest.md (hunk 2 above)
Comment (aboedo) [reply CHANGELOG.latest.md:1]: +1, let's get Amazon support first, then AdServices, then StoreKit 2 (StoreKit 2 is great but ultimately is under-the-hood stuff that we'll use on behalf of the dev to make stuff more reliable)

**Pull request merged**